### PR TITLE
Add token-cls task and sequence labeling demo

### PR DESCRIPTION
**Pull request by KPatr1ck**
@nepeplwu  @Steffy-zxf 

- Add MSRA dataset
- Add token_cls task and multi-classes case in ernie_tiny 
- Add sequence_labeling demo on MSRA

### CODE REVIEW

File: demo/sequence_labeling/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# PaddleHub Transformer模型fine-tune序列标注（动态图）
+
+本示例将展示如何使用PaddleHub Transformer模型（如 ERNIE、BERT、RoBERTa等模型）module 以动态图方式fine-tune并完成预测任务。
```
Comment (nepeplwu) [inline demo/sequence_labeling/README.md]: module建议首字母大写`Module`

File: demo/sequence_labeling/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    print(f'Data: {text} \t Lable: {", ".join(results[idx][1:len(text)+1])}')
+```
+
+参数配置正确后，请执行脚本`python predict.py`， 加载模型具体可参见[加载](https://www.paddlepaddle.org.cn/documentation/docs/zh/2.0-rc/api/paddle/framework/io/load_cn.html#load)。
+
+### 依赖
+
+paddlepaddle >= 2.0.0rc
+
+paddlenlp >= 2.0.0
```
Comment (nepeplwu) [inline demo/sequence_labeling/README.md]: 依赖paddlenlp的2.0.0beta版本还是正式版本？

File: paddlehub/datasets/base_nlp_dataset.py [hunk 3]
```diff
@@ -258,3 +259,139 @@ def __getitem__(self, idx):
...
+    def _convert_examples_to_records(self, examples: List[InputExample]) -> List[dict]:
+        """
+        Returns a list[dict] including all the input information what the model need.
+        Args:
+            examples (list): the data examples, returned by _read_file.
+        Returns:
+            a list with all the examples record.
+        """
+        records = []
+        with tqdm(total=len(examples)) as process_bar:
```
Comment (nepeplwu) [inline paddlehub/datasets/base_nlp_dataset.py]: 如果数据处理耗时不久的话，建议不加入tqdm的依赖

File: paddlehub/datasets/base_nlp_dataset.py [hunk 4]
```diff
@@ -258,3 +259,139 @@ def __getitem__(self, idx):
...
+                                self.label_list.index(labels[tokens_index]))
+                            tokens_index += 1
+                        else:
+                            record["label"].append(
+                                self.label_list.index(self.no_entity_label))
+                records.append(record)
+                process_bar.update(1)
+        return records
+
+    def _reseg_token_label(self, tokens, labels=None):
```
Comment (nepeplwu) [inline paddlehub/datasets/base_nlp_dataset.py]: token，labels缺少annotation。其他接口请再检查一遍

File: paddlehub/datasets/msra_ner.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            max_seq_len=max_seq_len,
+            mode=mode,
+            data_file=data_file,
+            label_file=None,
+            label_list=["B-PER", "I-PER", "B-ORG", "I-ORG", "B-LOC", "I-LOC", "O"],
+            is_file_with_header=True,
+        )
+
+
+if __name__ == "__main__":
```
Comment (nepeplwu) [inline paddlehub/datasets/msra_ner.py]: main相关的代码建议去除

File: demo/sequence_labeling/README.md (hunk 2 above)
Comment (nepeplwu) [reply demo/sequence_labeling/README.md]: 另外，requirements里面需要补充上paddlenlp的依赖

File: modules/text/language_model/ernie_tiny/module.py [hunk 6]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
-        # TODO(zhangxuefei): add token_classification task
+        if label_map:
+            self.num_classes = len(label_map)
+        else:
+            self.num_classes = num_classes
+
         if task == 'sequence_classification':
-            self.model = ErnieForSequenceClassification.from_pretrained(pretrained_model_name_or_path='ernie_tiny')
+            task = 'seq-cls'
+            warn(
```
Comment (nepeplwu) [inline modules/text/language_model/ernie_tiny/module.py]: 建议直接用hub内置的logger.warning来提示警告

File: demo/sequence_labeling/train.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    model = hub.Module(
+        name='ernie_tiny',
+        version='2.0.1',
+        task='token-cls',
+        label_map=label_map,
+    )
+
+    train_dataset = hub.datasets.MSRA_NER(
+        tokenizer=model.get_tokenizer(),
+        max_seq_len=50,
```
Comment (Steffy-zxf) [inline demo/sequence_labeling/train.py]: `max_seq_len`设置有点小。可以看看MSRA_NER数据集文本平均长度是多少

File: demo/sequence_labeling/train.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    )
+
+    dev_dataset = hub.datasets.MSRA_NER(
+        tokenizer=model.get_tokenizer(),
+        max_seq_len=50,
+        mode='dev'
+    )
+
+    optimizer = paddle.optimizer.AdamW(learning_rate=5e-5, parameters=model.parameters())
+    trainer = hub.Trainer(model, optimizer, checkpoint_dir='token_cls_save_dir', use_gpu=False)
```
Comment (Steffy-zxf) [inline demo/sequence_labeling/train.py]: 建议默认开启gpu训练

File: demo/sequence_labeling/train.py [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+    trainer.train(train_dataset, epochs=1, batch_size=16, eval_dataset=dev_dataset, save_interval=1)
```
Comment (Steffy-zxf) [inline demo/sequence_labeling/train.py]: `epoch`，`batch`设置也有点小。建议参考1.8分支下对应demo的config设置

File: modules/text/language_model/ernie_tiny/module.py [hunk 10]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
+        'seq-cls',
+        'token-cls',
+    ]
 
     def __init__(
             self,
             task=None,
             load_checkpoint=None,
             label_map=None,
+            num_classes=2,
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py:47]: 建议这里可以再增加一个**kwargs，支持用户透传参数

File: modules/text/language_model/ernie_tiny/module.py (hunk 6 above)
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: 使用`logger.warning(***)`，参考：https://github.com/PaddlePaddle/PaddleHub/blob/release/v2.0.0-beta/paddlehub/module/module.py#L270

File: modules/text/language_model/ernie_tiny/module.py [hunk 11]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
         if task == 'sequence_classification':
-            self.model = ErnieForSequenceClassification.from_pretrained(pretrained_model_name_or_path='ernie_tiny')
+            task = 'seq-cls'
+            warn(
+                "current task name 'sequence_classification' was renamed to 'seq-cls', "
+                "'sequence_classification' will be removed in the future",
+                DeprecationWarning,
+            )
+        if task == 'seq-cls':
+            self.model = ErnieForSequenceClassification.from_pretrained(pretrained_model_name_or_path='ernie_tiny', num_classes=self.num_classes)
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: `self.model = ErnieForSequenceClassification.from_pretrained(pretrained_model_name_or_path='ernie-tiny', num_classes=self.num_classes) `

paddlenlp已经将 ernie_tiny 重命名为 ernie-tiny

File: modules/text/language_model/ernie_tiny/module.py [hunk 12]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
+            self.criterion = paddle.nn.loss.CrossEntropyLoss()
+            self.metric = paddle.metric.Accuracy(name='acc_accumulation')
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: 这里去掉metric指定name吧。`paddle.metric.Accuracy()`

File: modules/text/language_model/ernie_tiny/module.py [hunk 13]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
+        elif task == 'token-cls':
+            self.model = ErnieForTokenClassification.from_pretrained(pretrained_model_name_or_path='ernie_tiny', num_classes=self.num_classes)
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: 同上

File: modules/text/language_model/ernie_tiny/module.py [hunk 14]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
             self.criterion = paddle.nn.loss.CrossEntropyLoss()
             self.metric = paddle.metric.Accuracy(name='acc_accumulation')
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: 同上

File: modules/text/language_model/ernie_tiny/module.py [hunk 15]
```diff
@@ -37,23 +38,44 @@ class ErnieTiny(nn.Layer):
...
         elif task is None:
             self.model = ErnieModel.from_pretrained(pretrained_model_name_or_path='ernie_tiny')
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py]: 同上

File: paddlehub/datasets/base_nlp_dataset.py [hunk 16]
```diff
@@ -258,3 +259,139 @@ def __getitem__(self, idx):
 
     def __len__(self):
         return len(self.records)
+
+
+class SeqLabelingDataset(BaseNLPDataset, fluid.io.Dataset):
```
Comment (Steffy-zxf) [inline paddlehub/datasets/base_nlp_dataset.py]: 去掉fluid，使用paddle.io.DataSet
其他代码如果有出现fluid，同样也去掉

加上相应的注释

**Review by Steffy-zxf — CHANGES_REQUESTED** [review]
同时，
1. 注意下paddlehub/module/下 关于transformer类模型的modeling 文件的删除。
2. 对于nlp支持finetune的module需要抽象出一个基类，公用`training_step` 和 `validation_step`的方法

File: demo/sequence_labeling/train.py (hunk 7 above)
Comment (KPatr1ck) [reply demo/sequence_labeling/train.py]: 根据1.8版本的教程，改成了128

File: demo/sequence_labeling/README.md (hunk 2 above)
Comment (KPatr1ck) [reply demo/sequence_labeling/README.md]: 已改为 2.0.0a4，`requirements.txt`中也添加了

File: requirements.txt [hunk 17]
```diff
@@ -16,3 +16,4 @@ tqdm
 visualdl >= 2.0.0
 # gunicorn not support windows
 gunicorn >= 19.10.0; sys_platform != "win32"
+paddlenlp >= 2.0.0a4
```
Comment (Steffy-zxf) [inline requirements.txt]: paddlenlp >= 2.0.0b

**Review by Steffy-zxf — COMMENTED** [review]
README 可以参考 这个教程 https://aistudio.baidu.com/aistudio/projectdetail/1283423， 把一些图文说明可以用上。

File: demo/sequence_labeling/README.md [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+### Step1: 选择模型
+```python
+import paddlehub as hub
+
+model = hub.Module(name='ernie_tiny', version='2.0.1', task='token-cls')
+```
+
+其中，参数：
+
+* `name`：模型名称，可以选择`ernie`，`ernie-tiny`，`bert_chinese_L-12_H-768_A-12`，`chinese-roberta-wwm-ext`，`chinese-roberta-wwm-ext-large`等。
```
Comment (Steffy-zxf) [inline demo/sequence_labeling/README.md]: bert系列 我记得 改了名称

File: demo/sequence_labeling/README.md [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+    print(f'Data: {text} \t Lable: {", ".join(results[idx][1:len(text)+1])}')
+```
+
+参数配置正确后，请执行脚本`python predict.py`， 加载模型具体可参见[加载](https://www.paddlepaddle.org.cn/documentation/docs/zh/2.0-rc/api/paddle/framework/io/load_cn.html#load)。
+
+### 依赖
+
+paddlepaddle >= 2.0.0rc
+
+paddlenlp >= 2.0.0a4
```
Comment (Steffy-zxf) [inline demo/sequence_labeling/README.md]: paddlenlp 依赖可以去掉 因为在requirements.txt里有

**Review by Steffy-zxf — COMMENTED** [review]
文本分类 和 序列标注readme 参考教程 https://aistudio.baidu.com/aistudio/projectdetail/1294333
把相应图文说明加上。

File: modules/text/language_model/ernie_tiny/module.py [hunk 20]
```diff
@@ -37,23 +37,44 @@ class ErnieTiny(nn.Layer):
...
-        # TODO(zhangxuefei): add token_classification task
+        if label_map:
+            self.num_classes = len(label_map)
+        else:
+            self.num_classes = num_classes
+
         if task == 'sequence_classification':
-            self.model = ErnieForSequenceClassification.from_pretrained(pretrained_model_name_or_path='ernie_tiny')
+            task = 'seq-cls'
+            logger.warning(
```
Comment (Steffy-zxf) [inline modules/text/language_model/ernie_tiny/module.py:58]: warning可以加上 Deprecated 字眼，强调下。

File: demo/sequence_labeling/README.md (hunk 18 above)
Comment (KPatr1ck) [reply demo/sequence_labeling/README.md]: 已改，和module名字同步

**Review by Steffy-zxf — APPROVED** [review]
LGTM
